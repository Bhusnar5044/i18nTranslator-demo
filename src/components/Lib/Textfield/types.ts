import { ChangeEvent } from 'react';

export type Props = {
    label: string;
    name?: string;
    value: any;
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
    placeholder?: string;
    disabled?: boolean;
};
