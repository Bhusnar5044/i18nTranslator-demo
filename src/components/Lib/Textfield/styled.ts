import { styled } from '@utils';

export const InputWrapper = styled.div`
    position: relative;
`;

export const Label = styled.label`
    position: absolute;
    font-size: 1rem;
    left: 0;
    top: 50%;
    transform: translateY(-50%);
    background-color: white;
    color: gray;
    padding: 0 0.3rem;
    margin: 0 0.5rem;
    transition: 0.1s ease-out;
    transform-origin: left top;
    pointer-events: none;
`;

export const Input = styled.input`
    font-size: 1rem;
    outline: none;
    border: 1px solid gray;
    border-radius: 5px;
    padding: 0.5rem 0.7rem;
    color: gray;
    transition: 0.1s ease-out;

    &:focus {
        border-color: #6200ee;
    }

    &:focus + label {
        color: #6200ee;
        top: 0;
        transform: translateY(-50%) scale(0.9);
    }

    &:not(:placeholder-shown) + label {
        top: 0;
        transform: translateY(-50%) scale(0.9);
    }
`;
