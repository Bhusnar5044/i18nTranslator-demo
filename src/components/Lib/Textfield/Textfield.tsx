import { FC, forwardRef, memo } from 'react';
import { Input, InputWrapper, Label } from './styled';
import { Props } from './types';

export const Textfield: FC<Props> = memo(
    forwardRef<HTMLInputElement, Props>(({ label, ...props }, ref) => {
        return (
            <InputWrapper>
                <Input ref={ref} type="text" {...props} />
                <Label>{label}</Label>
            </InputWrapper>
        );
    }),
);

Textfield.displayName = 'Textfield';
