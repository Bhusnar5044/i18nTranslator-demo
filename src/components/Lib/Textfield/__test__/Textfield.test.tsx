import { render } from '@test-utils';
import { Textfield } from '../Textfield';

describe('Textfield', () => {
    it('should render properly', () => {
        const { container } = render(<Textfield />);
        expect(container).toMatchSnapshot();
    });
});
