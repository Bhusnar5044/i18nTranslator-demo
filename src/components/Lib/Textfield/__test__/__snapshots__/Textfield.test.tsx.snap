// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textfield > should render properly 1`] = `
<div>
  <div
    class="sc-eLqIir cOViYd"
  >
    <input
      class="sc-hIFLYB iWjxZU"
      type="text"
    />
    <label
      class="sc-HzRMQ dZzpxc"
    />
  </div>
</div>
`;
